src/permissions_template.c

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright (C) 2015 The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file LICENSE for details.
+*/
+
+
+#include "kerneltypes.h"
+#include "kmalloc.h"
+#include "permissions_template.h"
+#include "console.h"
+
+#define MAX_NUMBER_OF_PERMISSIONS 169 // temporary, for now they'll be statically allocated
+
+static uint32_t template_counter = 1;
+static struct permissions_template* permissions_table[MAX_NUMBER_OF_PERMISSIONS]; // statically allocated array of templates
+
+uint32_t generate_new_identifier() {
+    return template_counter++;
```

Comment:
Currently, there is a permissions check function can be used to prevent access to non-owned templates. I looked into using a dictionary or something but the complexity didn't seem worth it. Do you think we should still do more?